lib/liquid/context.rb

```diff
@@ -74,32 +74,13 @@ def pop_interrupt
     end
 
     def handle_error(e, line_number = nil, raw_token = nil)
-      if e.is_a?(Liquid::Error)
-        e.template_name ||= template_name
-        e.line_number ||= line_number
-      end
-
-      output = nil
-
-      if exception_handler
-        args = [e]
-        args << { line_number: line_number, raw_token: raw_token } if exception_handler.arity == 2
```

Comment:
I feel like `raw_token` was added outside of the error for the same reason that `line_number` was added outside of the error, because we couldn't attach that information to the error.  We actually already have a Liquid::Error#markup_context which looks like it serves the same purpose.

The other problem that we have is that we automatically render information that is on the Error, which prevents us from getting line number information without displaying it.  We could override that in the liquid_renderer, but we have made it awkward to get the liquid error message without the context (e.g. `error.to_s(false)`).  Once we deal with that issue we could actually have the line number as well as the token, without exposing them to the world.